### PR TITLE
feat: :sparkles: reactify workspace-fusion setup

### PR DESCRIPTION
**Pull request by Gustav-Eikaas**
# Description

Please include a summary of the change and which issue is fixed.
List any external dependencies that are required for this change.

Completes: AB#FILL_IN_YOUR_ISSUE_ID

## Type of change

Remove options that are not relevant.

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Refactor (A code change that neither fixes a bug nor adds a feature)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist

-   [ ] My code follows the style guidelines of this project
-   [ ] I have performed a self-review of my own code.
-   [ ] I have linked my DevOps task using the AB# tag.
-   [ ] My code is easy to read.
-   [ ] Documentation and comments is added where needed.
-   [ ] My code is covered by tests.
-   [ ] UX has reviewed relevant UI contributions.

### CODE REVIEW

File: packages/workspace-fusion/src/lib/components/Workspace.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import { FilterConfig } from '../integrations/filter';
+import { GardenConfig } from '../integrations/garden';
+import { GridConfig } from '../integrations/grid';
+import { StatusBarConfig } from '../integrations/status-bar';
+import { SidesheetConfig } from '../integrations/sidesheet';
+import { FusionPowerBiConfig, PowerBiConfig } from '../integrations/power-bi';
+import { didOptionsChange } from '../utils/optionsChanged/didOptionsChange';
+import { createConfigurationObject } from '../utils/createWorkspaceConfig';
+
+type OnWorkspaceReadyEvent<TData extends Record<PropertyKey, unknown>> = {
```
Comment (Gustav-Eikaas) [inline packages/workspace-fusion/src/lib/components/Workspace.tsx:14]: export to types

File: packages/workspace-fusion/src/lib/types/configuration.ts [hunk 2]
```diff
@@ -29,3 +43,19 @@ export type AppConfig<TabNames extends string> = {
...
 	fetch: (uri: string, init?: RequestInit) => Promise<Response>;
 };
+
+export type WorkspaceConfiguration<
+	TData extends Record<PropertyKey, unknown>,
+	TContext extends Record<PropertyKey, unknown>,
+	TExtendedFields extends string = never,
+	TCustomGroupByKeys extends Record<PropertyKey, unknown> = never
+> = {
+	gardenController?: GardenController<TData, TExtendedFields, TCustomGroupByKeys, TContext, TContext>;
```
Comment (jaysencpp) [inline packages/workspace-fusion/src/lib/types/configuration.ts:53]: Why does GardenController take TContext twice as generic arguments?

File: packages/workspace-fusion/src/lib/types/fusionController.ts [hunk 3]
```diff
@@ -1,7 +1,11 @@
 import { WorkspaceReactMediator } from '@equinor/workspace-react';
-import { WorkspaceContext } from '../classes/fusionWorkspaceBuilder';
 import { FusionBookmark } from './fusionBookmark';
-import { WorkspaceOnClick } from './onClick';
+import { WorkspaceOnClick } from './event';
+
+//TODO: Generic, temp
```
Comment (jaysencpp) [inline packages/workspace-fusion/src/lib/types/fusionController.ts:5]: //TODO: Generic, temp
What does it mean?

File: packages/workspace-fusion/src/lib/utils/createWorkspaceConfig.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+		addStatusBar(props.statusBarOptions, viewController, mediator);
+	}
+
+	mediator.clickService.click$.subscribe(({ item }) => {
+		const id = props.workspaceOptions.getIdentifier(item);
+		mediator.selectionService.selectedNodes = [id];
+		updateQueryParams([`item=${id}`], mediator, history);
+	});
+
+	//do it later
```
Comment (jaysencpp) [inline packages/workspace-fusion/src/lib/utils/createWorkspaceConfig.ts:91]: Add a `TODO:` so it's easier to see in the vscode extension ;)

File: packages/workspace-fusion/src/lib/utils/optionsChanged/didOptionsChange.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { WorkspaceProps } from '../../components';
+import { WorkspaceConfiguration } from '../../types';
+import { didDataSourceOptionsChange } from './dataSourceOptionsChanged.ts/dataSourceChanged';
```
Comment (jaysencpp) [inline packages/workspace-fusion/src/lib/utils/optionsChanged/didOptionsChange.ts:3]: Any reason why you called a folder for dataSourcesOptionsChanges.ts?

Comment (Gustav-Eikaas) [reply packages/workspace-fusion/src/lib/utils/optionsChanged/didOptionsChange.ts:3]: Naming is hard

Comment (Gustav-Eikaas) [reply packages/workspace-fusion/src/lib/utils/optionsChanged/didOptionsChange.ts:3]: Will fix

**Review by Noggling — APPROVED** [review]

**Pull request merged**